mirrord-layer/src/pod_api.rs

```diff
@@ -467,13 +480,21 @@ async fn wait_for_agent_startup(
         )
         .await?;
 
-    while let Some(line) = logs.try_next().await? {
-        let line = String::from_utf8_lossy(&line);
-        if line.contains("agent ready") {
-            break;
+    let logs_wait = tokio::spawn(async move {
+        while let Some(line) = logs.try_next().await? {
+            let line = String::from_utf8_lossy(&line);
+            if line.contains("agent ready") {
+                break;
+            }
         }
+
+        Ok(())
+    });
+
+    tokio::select! {
+        log_wait = logs_wait => log_wait?,
+        _ = tokio::time::sleep(Duration::from_secs(communication_timeout.unwrap_or(30) as u64)) => Err(LayerError::AgentReadyTimeout),
```

Comment:
when I wrote wrap all I meant the whole "startup" function for the agent: creating of the job, and waiting for the logs. since it could get stuck on the agent getting ready

Reply:
I see what you mean, simply moving to the spawn makes the borrow checker not very happy with lifetimes so I'm making the necessary clones and such